addons/mass_mailing/models/mailing.py

```diff
@@ -349,6 +381,31 @@ def action_duplicate(self):
             }
         return False
 
+    def action_edit_mailing(self):
+        self.ensure_one()
+        return {
+            'name': _('New Mailing Test'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'mailing.mailing',
+            'res_id': self.id,
+            'context': dict(self.env.context, edit=True, form_view_initial_mode='edit')
+        }
+
+    def action_select_as_winner(self):
+        self.ensure_one()
+        if self.testing_mailing_id:
+            final_mailing = self.copy()
```

Comment:
I'm pretty sure "copy" accepts default values so you could do

`final_mailing.copy({version: 'blah', 'is_winner': True})`
To avoid creating a record and then directly updating.